client/components/layout/main-layout/MainLayout.tsx

```diff
@@ -1,24 +1,30 @@
 import React from 'react';
 import { Outlet } from 'react-router-dom';
-import Header from './header/Header';
 import Footer from './footer/Footer';
+import LinktaLogo from './header/LinktaLogoWithText';
+import TopNavigationBar from './header/TopNavigationBar';
+import useDynamicNavigation from '@/client/hooks/useDynamicNavigation';
 /**
- * MainLayout component wraps the children components with a common structure
- * It is used to provide a consistent layout across different pages.
- * @param props - The component props.
- * @param props.showFooter - Determines if the footer should be rendered. Defaults to false.
- * @returns The MainLayout component with Header, Outlet for nested routes, and optional Footer.
+ * Provides a consistent layout structure across the app with conditional
+ * rendering of the top navigation bar and footer based on the current route.
  */
 interface MainLayoutProps {
   showFooter?: boolean;
 }
 
 const MainLayout: React.FC<MainLayoutProps> = ({ showFooter = false }) => {
+  const { showTopNavBar } = useDynamicNavigation();
+
   return (
     <div>
       <header>
-        <Header />
+        <LinktaLogo />
       </header>
+      {showTopNavBar && (
+        <nav>
+          <TopNavigationBar />
+        </nav>
+      )}
```

Comment:
Little curious about this - `TopNavigationBar` is located within the header directory, so I imagine it's considered part of the header, but the `<nav>` element is rendered outside of the `<header>` element. Did you have particular considerations around this like styling?

Reply:
Yes, I moved the nav out of the header because it is only needed for certain pages and not others according to our mockup.